pkg/graph/resolvers/it_gov_task_statuses_test.go

```diff
@@ -646,14 +653,215 @@ func TestBizCaseDraftStatus(t *testing.T) {
+		},
+		{
+			testCase: "GRT Step: GRT Date Tommorrow",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepGRTMEETING,
+				GRTDate: &tomorrow,
+			},
+			expectedStatus: models.ITGGRTSScheduled,
+			expectError:    false,
+		},
+		{
+			testCase: "Final Business Case Step: No GRT Date Scheduled",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepFINALBIZCASE,
+				GRTDate: nil,
+			},
+			expectedStatus: models.ITGGRTSNotNeeded,
+			expectError:    false,
+		},
+		{
+			testCase: "Final Business Case Step: GRT Date Yesterday",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepFINALBIZCASE,
+				GRTDate: &yesterday,
+			},
+			expectedStatus: models.ITGGRTSCompleted,
+			expectError:    false,
+		},
+		{
+			testCase: "Final Business Case Step: GRT Date Tommorrow",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepFINALBIZCASE,
+				GRTDate: &tomorrow,
+			},
+			expectedStatus: models.ITGGRTSScheduled,
+			expectError:    false,
+		},
+		{
+			testCase: "GRB Step: No GRT Date Scheduled",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepGRBMEETING,
+				GRTDate: nil,
+			},
+			expectedStatus: models.ITGGRTSNotNeeded,
+			expectError:    false,
+		},
+		{
+			testCase: "GRB Step: GRT Date Yesterday",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepGRBMEETING,
+				GRTDate: &yesterday,
+			},
+			expectedStatus: models.ITGGRTSCompleted,
+			expectError:    false,
+		},
+		{
+			testCase: "GRB Step: GRT Date Tommorrow",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepGRBMEETING,
+				GRTDate: &tomorrow,
+			},
+			expectedStatus: models.ITGGRTSScheduled,
+			expectError:    false,
+		},
+		{
+			testCase: "Decision Step: No GRT Date Scheduled",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepDECISION,
+				GRTDate: nil,
+			},
+			expectedStatus: models.ITGGRTSNotNeeded,
+			expectError:    false,
+		},
+		{
+			testCase: "Decision Step: GRT Date Yesterday",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepDECISION,
+				GRTDate: &yesterday,
+			},
+			expectedStatus: models.ITGGRTSCompleted,
+			expectError:    false,
+		},
+		{
+			testCase: "Decision Step: GRT Date Tommorrow",
+			intake: models.SystemIntake{
+				Step:    models.SystemIntakeStepDECISION,
+				GRTDate: &tomorrow,
+			},
+			expectedStatus: models.ITGGRTSScheduled,
+			expectError:    false,
+		},
+		{
+			testCase: "Decision Step: No GRT Date Scheduled",
+			intake: models.SystemIntake{
+				Step:    defaultTestStep,
+				GRTDate: nil,
+			},
+			expectedStatus: "",
+			expectError:    true,
+		},
+		{
+			testCase: "Decision Step: GRT Date Yesterday",
+			intake: models.SystemIntake{
+				Step:    defaultTestStep,
+				GRTDate: &yesterday,
+			},
+			expectedStatus: models.ITGGRTSCompleted,
+			expectError:    false,
+		},
+		{
+			testCase: "Decision Step: GRT Date Tommorrow",
```

Comment:
This test case needs a different name that mentions that the step is invalid, but there's no error.

Reply:
fixed!